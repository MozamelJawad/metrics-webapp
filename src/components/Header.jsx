import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <div className="header">
    <Link to="/">
      <i className="bi bi-arrow-left-short" />
    </Link>
    <div className="header-icon">
      <i className="bi bi-mic-fill" />
      <i className="bi bi-gear-fill" />
    </div>
  </div>
);

export default Header;
