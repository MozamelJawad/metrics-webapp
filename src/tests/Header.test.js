import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';
import '@testing-library/jest-dom';

describe('Header: ', () => {
  it('Should render the header correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('Navigation link work correctly', () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const homeLink = screen.getByTestId('home');
    expect(homeLink).toBeInTheDocument();
    fireEvent.click(getByTestId('home'));
  });
});
