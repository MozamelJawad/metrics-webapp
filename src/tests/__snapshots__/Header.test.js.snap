// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header:  Should render the header correctly 1`] = `
<div>
  <div
    class="header"
  >
    <a
      data-testid="home"
      href="/"
    >
      <i
        class="bi bi-arrow-left-short"
      />
    </a>
    <div
      class="header-icon"
    >
      <i
        class="bi bi-mic-fill"
      />
      <i
        class="bi bi-gear-fill"
      />
    </div>
  </div>
</div>
`;
